feat: [plugin-sui] programmable transaction block generation action

emeieiron opened the pull request:
<!-- Use this template by filling in information and copy and pasting relevant items out of the html comments. -->

# Relates to:

<!-- LINK TO ISSUE OR TICKET -->

<!-- This risks section is to be filled out before final review and merge. -->

# Risks
- no risks

<!--
Low, medium, large. List what kind of risks, and what could be effected.
-->

# Background

## What does this PR do?
Adds a Programmable Transaction Block Builder Action to Sui Plugin

## What kind of change is this?
Features (non-breaking change which adds functionality)

<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->

<!-- This "Why" section is most relevant if there is no linked issue explaining why. If there is a related issue it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->

# Documentation changes needed?

<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If a docs change is needed: I have updated the documentation accordingly.
-->

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested, and probably help the PR get merged quicker. -->

# Testing

## Where should a reviewer start?
- `cd` into `packages/plugin-sui`
- run `pnpm i`

## Detailed testing steps
- Replace Default Character [here](https://github.com/elizaOS/eliza/blob/051323f3a550b697c3ea3212023da35b191d3287/agent/src/index.ts#L486):
```
    // Replace the existing character initialization with custom setup
    const character = defaultCharacter;

    character.plugins = [suiPlugin];
    character.modelProvider = ModelProviderName.OPENAI;
    let characters = [character];
```

<!--
None, automated tests are fine.
-->

<!--
- As [anon/admin], go to [link]
  - [do action]
  - verify [result]
-->

<!-- If there is a UI change, please include before and after screenshots or videos. This will speed up PRs being merged. It is extra nice to annotate screenshots with arrows or boxes pointing out the differences. -->
<!--
## Screenshots
### Before
### After
-->

<!-- If there is anything about the deploy, please make a note. -->
<!--
# Deploy Notes
-->

<!--  Copy and paste commandline output. -->
<!--
## Database changes
-->

<!--  If there is something more than the automated steps, please specifiy deploy instructions. -->
<!--
## Deployment instructions
-->

<!-- If you are on Discord, please join https://discord.gg/ai16z and state your Discord username here for contribute role and join us in #development-feed -->

## Discord username
@em.eieiron


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
	- Added a new action `buildPTB` to the Sui plugin
	- Introduced advanced transaction block (PTB) building capabilities
	- Implemented schema validation for complex transaction commands

- **Improvements**
	- Enhanced plugin functionality with more flexible transaction handling
	- Added robust input validation for transaction block creation

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

astinz commented:
POW

![CleanShot 2024-12-22 at 19 27 28@2x](https://github.com/user-attachments/assets/00f69dad-34f1-4c5e-8606-af81e2cef89c)

odilitime commented:
not sure what's up with the smoke-test

astinz commented:
must've been the differences in `pnpm-lock.yaml`. a rebase seems to have cleaned it.

0xnogo committed:
feat: Add new translations and update README structure

- Introduced new language translations for Eliza in Arabic, Chinese, German, Spanish, French, Hebrew, Hungarian, Italian, Japanese, Korean, Dutch, Polish, Portuguese, Romanian, Russian, Serbian, Tagalog, Thai, Turkish, and Vietnamese.
- Updated the README structure to include links to the new translations.
- Added a new plugin '@elizaos/plugin-hyperliquid' in the pnpm-lock.yaml file with its dependencies.
- Adjusted the README links to point to the correct paths for translations.

HashWarlock committed:
fix unit tests and update package json

shakkernerd committed:
Merge pull request #2303 from Phala-Network/fix/ra-and-derive-key

fix: derive key and update remote attestation

wtfsayo committed:
Merge branch 'develop' into direct-client_agent-crud

wtfsayo committed:
Merge pull request #2267 from jseparovic/direct-client_agent-crud

Improvement: Direct Client API - Add Delete Agent functionality

francis2tm committed:
feat: add new model classes for embedding and image processing

francis2tm committed:
feat: extend handleProvider function to include runtime, context, and modelClass options

CryptoGraffe committed:
fix farcaster memory

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

shakkernerd committed:
chore: console -> elizaLogger

lalalune committed:
Update README.md

lalalune committed:
Merge pull request #2309 from elizaOS/lalalune-patch-2

Update README.md

shakkernerd committed:
Merge branch 'develop' into pr/CryptoGraffe/2307

wtfsayo committed:
Merge branch 'develop' into refactor-farcaster-client

wtfsayo committed:
Merge branch 'develop' into 1186-add-get-memories-by-ids

wtfsayo committed:
Merge pull request #2293 from bbopar/1186-add-get-memories-by-ids

feat: add getMemoryByIds to database adapters

wtfsayo committed:
Merge branch 'develop' into character-from-url

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update .env.example

wtfsayo committed:
Merge pull request #2281 from v1xingyue/character-from-url

feat: support load character from  character_url

shakkernerd committed:
Merge branch 'develop' into pr/augchan42/2248

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2248 from augchan42/feature/echochamber-conversation-starter

feat: (echochambers) add dead room detection and conversation starter

wtfsayo committed:
chore: Update README.md (#2280)

* Update README.md

---------

Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

0xnogo committed:
moving new locales

0xnogo committed:
Remove swipr from git tracking

sin-bufan committed:
Merge branch 'develop' into refactor-farcaster-client

Lukapetro committed:
fix: remove goku fallback image, env variable renamed

Lukapetro committed:
chore: resolve merge conflicts with develop

Commit:
feat: support for eternalai provider can get onchain system prompt

Commit:
Merge remote-tracking branch 'origin/develop' into eternalai-agent

# Conflicts:
#	.env.example

shakkernerd committed:
Merge pull request #2087 from sin-bufan/refactor-farcaster-client

refactor: farcaster client env configuration

Lukapetro committed:
chore: resolve merge conflicts with develop

Commit:
Merge branch 'elizaOS:develop' into eternalai-agent

Commit:
feat: support for eternalai provider can get onchain system prompt

Commit:
Merge remote-tracking branch 'origin/eternalai-agent' into eternalai-agent

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

shakkernerd committed:
Merge pull request #2319 from eternalai-org/eternalai-agent

feat: Onchain Agent Transformer - transform any Eliza agents into unstoppable Solidity smart contracts deployed on 10+ blockchains

francis2tm committed:
feat: add Atoma SDK configuration to .env.example for Bearer Auth token

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

wtfsayo committed:
Merge branch 'develop' into pr/1865

wtfsayo committed:
Delete pnpm-lock.yaml

wtfsayo committed:
Create pnpm-lock.yaml

wtfsayo committed:
Update index.ts

wtfsayo committed:
Update tokenProvider.ts

wtfsayo committed:
Update tokenProvider.ts

wtfsayo committed:
Merge pull request #1865 from hellopleasures/develop

feat: add Dex Screener plugin with token price action, evaluators, an…

tcm390 committed:
Merge branch 'develop' into refactor-websearch

wtfsayo committed:
Merge pull request #2195 from chuasonglin1995/refactor-websearch

refactor: websearch into a service

everimbaq committed:
fix: correct OPENROUTER_API_KEY env in config

Lukapetro committed:
Merge branch 'develop' into feature/add-client-instagram

shakkernerd committed:
Merge pull request #2324 from xNomad-AI/main

fix: correct OPENROUTER_API_KEY env in config

leeran7 committed:
Feat: Enables multiple remote urls

leeran7 committed:
Chore: Remove unneeded spaces

leeran7 committed:
Merge branch 'develop' into multiple-remote-urls

Lukapetro committed:
Merge branch 'develop' into feature/add-client-instagram

francis2tm committed:
feat: integrate Atoma model support with configuration updates and SDK version change

- Added Atoma API URL and model configurations to .env.example.
- Updated pnpm-lock.yaml to reference the latest Atoma SDK version.
- Enhanced index.ts to include Atoma as a model provider for token retrieval.
- Implemented Atoma model initialization and response handling in generation.ts.
- Updated models.ts to utilize dynamic settings for Atoma model classes.

francis2tm committed:
refactor: remove unused model configurations from .env.example and models.ts

- Deleted EMBEDDING_ATOMA_MODEL and IMAGE_ATOMA_MODEL from .env.example.
- Removed corresponding model class settings from models.ts to streamline configuration.

sukrucildirr committed:
Update README_TR.md

derrekcoleman committed:
Fix formatting in Browser section of README.md

Hey all! I saw strange formatting in the README.md and figured it was an extra three backticks ```

Sure enough, deleting those made the section render as intended. I also reworded one of the sentences that had three "and"s to make it a little more legible.

JoeyKhd committed:
upd: added initial docker image fixes

JoeyKhd committed:
upd: we need all packages in Dockerfile workdir

rferrari committed:
feat: groq image vision provider

JoeyKhd committed:
upd: reworked docker ignore

JoeyKhd committed:
upd: rebuilt docs and bumped version numbers

francis2tm committed:
Merge remote-tracking branch 'upstream/main' into feat/atoma-provider

francis2tm committed:
chore: update lockfile

francis2tm committed:
chore: update lockfile

francis2tm committed:
chore: update lockfile

ai16z-demirix committed:
Merge branch 'develop' into feature/solana-plugin-tests

JoeyKhd committed:
upd: resolved Dockerfile.docs build issues

JoeyKhd committed:
upd: reverted org generation in api docs

JoeyKhd committed:
upd: bumped types packages for docusaurus

thomasWos committed:
Update README.md

francis2tm committed:
refactor: restructure Atoma model configuration for improved clarity and consistency

- Updated Atoma model settings in models.ts to encapsulate parameters (stop, maxInputTokens, maxOutputTokens, temperature) within each model class.
- Reformatted code for better readability and maintainability.
- Minor formatting adjustments in types.ts for consistency in enum declaration and interface method signatures.

shakkernerd committed:
Merge branch 'develop' into pr/JoeyKhd/2347

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #2347 from JoeyKhd/fix-dockerimage

fix: docker images+compose and broken documentation

francis2tm committed:
chore: restore original formatting

francis2tm committed:
chore: restore original formatting

francis2tm committed:
chore: restore original formatting

francis2tm committed:
chore: restore original formatting

francis2tm committed:
chore: revert types.ts

francis2tm committed:
Merge remote-tracking branch 'upstream/develop' into feat/atoma-provider

francis2tm committed:
refactor: improve code formatting and readability in generation.ts

- Standardized import statements and removed unnecessary line breaks.
- Simplified function definitions and improved inline comments for clarity.
- Enhanced logging messages for better debugging and error tracking.
- Ensured consistent use of single quotes for strings throughout the file.

francis2tm committed:
refactor: remove Atoma model generation function from generation.ts

- Deleted the handleAtoma function, which was responsible for generating objects for Atoma models.
- This change simplifies the codebase by removing unused functionality, improving maintainability.

francis2tm committed:
chore: revert deps

francis2tm committed:
chore: remove atoma-sdk

mcp97 committed:
Merge branch 'develop' into patch-2

mcp97 committed:
Merge pull request #2346 from thomasWos/patch-2

Fix github client README.md

mcp97 committed:
Merge pull request #2342 from rferrari/feature/groq-image-vision-provider

feat: groq image vision provider

mcp97 committed:
Merge pull request #2220 from ai16z-demirix/fix/issue-2164

improvement: using strict types to avoid erorrs like issue 2164

augchan42 committed:
feat: (ragKnowledge) Enhance RAG knowledge handling

- Added support for double-byte characters (e.g., Chinese) to ensure proper processing.
- Implemented cleanup of deleted knowledge files to maintain data integrity.
- Enabled loading knowledge from directories, simplifying configuration and management.

These improvements address performance and usability issues, facilitating better support for large knowledge sets.

Closes #2323

augchan42 committed:
Merge branch 'develop' into feature/rag-knowledge-improvements-v2

tcm390 committed:
Merge pull request #2336 from derrekcoleman/patch-1

Fix formatting in Browser section of README.md

tcm390 committed:
Merge pull request #2334 from sukrucildirr/develop

Update README_TR.md

C0ldSmi1e committed:
fix: align base url in client if api runs on a different port

odilitime committed:
Merge branch 'develop' into feature/rag-knowledge-improvements-v2

odilitime committed:
Merge pull request #2351 from augchan42/feature/rag-knowledge-improvements-v2

feat: (ragKnowledge) Enhance RAG knowledge handling

odilitime committed:
add back farcaster

HashWarlock committed:
fix: esure RA Action reply does not hallucinate

omahs committed:
Update README_FR.md

0xnogo committed:
Merge branch 'develop' into ng/moving-readmes

odilitime committed:
Merge pull request #1964 from Lukapetro/feature/add-client-instagram

feat: add instagram client

odilitime committed:
Merge pull request #2356 from omahs/patch-2

docs: Update README_FR.md

odilitime committed:
Merge pull request #2355 from elizaOS/HashWarlock-fix-ra-action-reply

fix: esure RA Action reply does not hallucinate

odilitime committed:
Merge pull request #2353 from C0ldSmi1e/develop

fix: align base url in client if api runs on a different port

odilitime committed:
Merge branch 'develop' into ng/moving-readmes

proteanx committed:
feat: add safe_mode (& cfg_scale) for venice image generation (#2354)

* venice: add safe_mode & cfg_scale for images

* add missing types

for safeMode & cfgScale

* update lockfile

---------

Co-authored-by: Sayo <hi@sayo.wtf>

kesar committed:
Update package.json with proper dependencies (#2301)

these packages are being used in prod build

Co-authored-by: Sayo <hi@sayo.wtf>

odilitime committed:
Merge pull request #2149 from 0xnogo/ng/moving-readmes

chore: Reorganizing README translations into a dedicated i18n directory structure

odilitime committed:
bump version

odilitime committed:
bump lockfile

odilitime committed:
Merge pull request #2359 from elizaOS/odi-fix

chore: Prep 0.1.9-alpha.1

JoeyKhd committed:
Merge pull request #4 from elizaOS/develop

merge dev

odilitime committed:
fix lint issues

odilitime committed:
Merge pull request #2328 from leeran7/multiple-remote-urls

feat: Enable Multiple remote character urls

odilitime committed:
Merge pull request #2368 from elizaOS/odi-fix

chore: lint

defitricks committed:
fix: Fix typo in import statement for Solana AgentKit plugin Update i… (#2370)

* fix: Fix typo in import statement for Solana AgentKit plugin Update index.ts

* Update index.ts

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>

0xCardiE committed:
feat: dexscreener trending (#2325)

* add to main index

* cleanup

---------

Co-authored-by: Sayo <hi@sayo.wtf>

JoeyKhd committed:
upd: added the possibility to configure S3 endpoint, whether SSL is enabled and the forceStylePath

tcm390 committed:
check if remote url is defined first (#2384)

0xSero committed:
Startup Error: ENV misconfig (#2378)

* fix

* Update info.json

---------

Co-authored-by: Sero <69639595+Seroxdesign@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

dev-whoan committed:
fix: check env REMOTE_CHARACTER_URLS is defined (#2382)

Co-authored-by: Sayo <hi@sayo.wtf>

ericlehong committed:
Fix typo (#2385)

* fix: typo

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>

* fix: typo

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>

---------

Signed-off-by: ericlehong <193237094+ericlehong@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

Archethect committed:
fix: set default Squid Router throttle to 1000 (#2386)

crStiv committed:
chore: improve descriptions  (#2394)

* Update index.md

* Update quickstart.md

* Update agents.md

* Update actions.md

* chore: with -> through

shakkernerd committed:
Merge branch 'develop' into feature/solana-plugin-tests

shakkernerd committed:
Merge pull request #2345 from ai16z-demirix/feature/solana-plugin-tests

feat: adding tests for plugin-solana

Commit:
enhancement: invoke exit code 0 after positive cleanup

shakkernerd committed:
Merge branch 'develop' into pr/francis2tm/2082

shakkernerd committed:
Merge pull request #2082 from francis2tm/feat/atoma-provider

feat: atoma provider

shakkernerd committed:
Merge branch 'develop' into cleanup-exit-code

shakkernerd committed:
Merge pull request #2398 from erise133/cleanup-exit-code

chore: invoke exit code 0 after positive cleanup

shakkernerd committed:
feat: add eslint config file

shakkernerd committed:
fix: let -> const

shakkernerd committed:
fix: remove unused error var

shakkernerd committed:
fix: unused declaration of var

shakkernerd committed:
Merge pull request #2400 from elizaOS/fixes

fix: lint errors

ai16z-demirix committed:
feat: adding tests for slack client. Moving existing tests to new __tests__ directory.

ai16z-demirix committed:
feat: adding tests for github client

shakkernerd committed:
Merge pull request #2404 from ai16z-demirix/test/client-slack

feat: adding tests for slack client. Moving existing tests to new __tests__ directory.

shakkernerd committed:
Merge branch 'develop' into test/client-github

odilitime committed:
Merge pull request #2379 from JoeyKhd/s3-improvement

feat: better S3 flexibility

shakkernerd committed:
Merge pull request #2407 from ai16z-demirix/test/client-github

feat: adding tests for github client

enigmarikki committed:
feat: injective plugin (#1764)

* "plugin: init injective integration and skeleton"

* "wip : skeleton partially done"

* "wip: finished fetch"

* "chore : fix format"

* "chore : renamed files"

* "wip: completed grpc endpoints for onchain queries"

* "chore : removed unwanted files"

* "wip : adding templates + integrating onchain functions"

* "wip : added inital version of templates"

* "wip : testing the plugin"

* "wip : skeleton for actions, providers"

* "chore: renamed and fixed relative imports"

* "wip : added message broadcasting rpcs"

* "feat : finished initial version of message parsing and querying"

* "chore : refactor package.json"

* "chore: added chain message broadcasters to the InjectiveGrpcBase"

* "wip : restructure actions and providers"

* "chore : fix conflicts on lockfile"

* "chore: added readme.md to use injective-sdk-client-ts"

* "chore : renamed the package to ensure consistency and updated the docs"

* "chore : refactored function args to make it easy to define function schema"

* "fix : missing imports for wasm"

* "wip : refactor injective types and templates"

* "chore : refactored the types into different files for maintainability"

* "chore : fmt"

* "wip : refactor the responses for standard message parsing"

* "chore : refactored all modules to ouput same a common message type"

* "wip : minimal tests"

* "fix: the grpc init + bug in making request"

* "wip : done with skeleton templates"

* "chore : reorgranize files"

* "wip : adding exchange templates"

* "chore : added fmt and finished comprehensive template for exchange module"

* "wip: refactored templates for auction and bank"

* "wip : almost done refactoring all the templates to add both the request, response"

* "chore : renamed to gov for consistency and added the explorer module"

* "fix : removed minor import into exchange"

* "wip : added the base action"

* "feat: finished integrating actions, wip fix for templates and examples"

* "chore : refactored all the request and response templates"

* "chore : fmt templates"

* "chore : added modules examples except exchange, fixed a few relative imports"

* "chore: finished integrating exchange examples"

* "feat: finished all integrations for injective-plugin"

* "fix : revert fmt"

* "chore : bump injective-ts sdk version"

* "fix : dependency issues"

* "fix : complex dependency issue from esm imports"

* "chore : refactored the rpc to take either inj address or eth address"

* "wip: fixed rpc initialization"

* "fix: Refactored action template"

* "chore : resolve lock file conflicts"

* "chore : added config params in .env.examples"

* "wip : improve response prompts from agent"

* "chore : refactored similes, for better interpretation"

* "wip: integrated mito's constant product market maker - amm"

* "feat: added Initial Dex offering and action, for agents to autonomously raise funds"

* "chore : revert defaultCharacter to LLAMA"

---------

Co-authored-by: rikki <rikki@mahishasuramardini.mynetworksettings.com>
Co-authored-by: Odilitime <janesmith@airmail.cc>

its-DeFine committed:
feat: Integrate Livepeer LLM provider  (#2154)

* add livepeer on index.ts as llm provider

* updated livepeer models

* add livepeer as llm provider

* add retry logic on livepeer img gen

* add handlelivepeer

* update test

* add livepeer model keys on .example.env

* Merge pull request #2 from Titan-Node/livepeer-doc-updates

Updated docs for Livepeer LLM integration

* add endpoint on livepeer on models.ts

* edit livepeer model config at model.ts

* Add Livepeer to image gen plugin environments

Fixes this error
```
Error handling message: Error: Image generation configuration validation failed:
: At least one of ANTHROPIC_API_KEY, NINETEEN_AI_API_KEY, TOGETHER_API_KEY, HEURIST_API_KEY, FAL_API_KEY, OPENAI_API_KEY or VENICE_API_KEY is required
    at validateImageGenConfig (file:///root/eliza-test/eliza-livepeer-integration/packages/plugin-image-generation/dist/index.js:38:19)
```

* add comments on livepeer model sizes

* remove retry logic from livepeer generate text and img

* Fixed .env naming convention and fixed mismatch bug within code

* add bearer on livepeer calls

* change in parsing to accomodate for new livepeer update

* addadd nineteen api key on the message

---------

Co-authored-by: Titan Node <hello@titan-node.com>

wtfsayo committed:
Delete tate.character.json (#2425)

twilwa committed:
refactor: dockerize smoke tests (#2420)

* add workflow to block minified JS

* modify smoketests to run in a container, add caching

* use bullseye for bash

* node 23

---------

Co-authored-by: Sayo <hi@sayo.wtf>

sweetmantech committed:
feat: coinbase agentkit plugin for eliza (#2298)

* I add files for plugin-agentkit

* instructions added to gitignore.

* I update package name.

* I add updated files.

* I add ENV requirements to prevent default Eliza bugs.

* rollback index

* simple agentkit import.

* I load existing wallet data for provider.

* remove dev logging

* Update pnpm-lock.yaml

* update .env.example and support non 'base-sepolia'

* Update pnpm-lock.yaml

* Update package lock for smoke tests.

* I rollback changes to pnpm-lock file.

* fixing smoke tests for pnpm missing

* rollback changes to agent start file.

* I rollback changes to smoke test.

* Update pnpm-lock.yaml

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

twilwa committed:
feat: add workflow to block minified JS (#2417)

* add workflow to block minified JS

* Update block-mini.yml

print violating filenames and line numbers

* Modify threshold

zo-sol committed:
feat: Code In Plugin, load characters from blockchain (#2371)

* add ia

* Delete pnpm-lock.yaml

* Update agent/src/index.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* Update packages/plugin-iq6900/package.json

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* Update agent/src/index.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* update solana version as others

* Update package.json

* test iq6900 env issue

* fix env file and lock

* edited require

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

YoungPhlo committed:
docs: Add notes for weekly contributor meeting on 2025-01-14 (#2426)

Co-authored-by: Sayo <hi@sayo.wtf>

tcm390 committed:
fix: Fix Incorrect Tweet ID Parameter Passed to sendTweet Function (#2430)

* correct id

* skip action if model is not supported

* warn

* update warn

wtfsayo committed:
Update quickstart.md

tcm390 committed:
Merge pull request #2437 from elizaOS/chore-remove-merge-conflict-in-quickstart

fix: quickstart.md conflicts

Noisyxl committed:
Update README.md (#2438)

leopardracer committed:
Update seed.sql (#2435)

Co-authored-by: Sayo <hi@sayo.wtf>

AIFlowML committed:
feat: Pyth Data Plugin  (#2434)

* feat(pyth): add Pyth Data plugin

- Create new plugin for Pyth Data integration
- Add EventSource client for price feeds
- Configure plugin in agent
- Update dependencies and configuration files

This plugin enables real-time price feed data from Pyth Network,
including price updates, TWAPs, and publisher caps information.

* feat: Wrap up changes for Pyth Data integration

* Save local changes before merging upstream

* Update index.ts, package.json, and add .env.example for Pyth Data integration

* Missing lock

---------

Co-authored-by: Sayo <hi@sayo.wtf>

JoeyKhd committed:
upd: if VITE_SERVER_PORT is not defined, use default 3000 (#2433)

Co-authored-by: Sayo <hi@sayo.wtf>

Hopium21 committed:
Update README.md (#2441)

Co-authored-by: Sayo <hi@sayo.wtf>

KacperKoza343 committed:
feat: ibc transfer on cosmos blockchains (#2358)

* feat: ibc transfer action

* refactor: remove redundant services and use skipClient

* update: eliza package import

* feat: add bridge data fetcher and update tests

* update: add IBC transfer action to README

* refactor: make bridge data fetcher reusable across the project

* fix: update similes for ibc transfer action

* fix: update toAddress in ibc transfer

* refactor: add changes according to review

* fix: ibc transfer action

bundinho committed:
fix TEE Log plugin errors at agent startup (#2415)

* fix TEE Log configuration error

* fix error telling 'Cannot open database because the directory does not exist'

* fix error telling 'Invalid TEE_MODE: undefined. Must be one of: LOCAL, DOCKER, PRODUCTION' at start up

wtfsayo committed:
Update index.ts (#2444)

JoeyKhd committed:
feat: improvement to logger (#2396)

* upd: initial updated logger

* upd: remove test case

* upd: improve some logs

* upd: resolved issues with mix of left and right json

* fix env and default mode

* Update pnpm-lock.yaml

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>

tsubasakong committed:
feat:update heurist env var examples (#2428)

Co-authored-by: Sayo <hi@sayo.wtf>

RubinovaAn1097 committed:
Fix typos in docs (#2449)

* Update MemoryCacheAdapter.md

* Update index.md

MarsonKotovi4 committed:
fix: Update error message and remove duplicate version field (#2445)

* Update index.ts

* Update package.json

---------

Co-authored-by: Sayo <hi@sayo.wtf>

ae9is committed:
fix: #2373 Fix image description (#2375)

* fix: gif frame extraction and remove outdated dependency "gif-frames"

* fix: local image description

* fix: describe image action to account for differences in file location object between models

* fix: image description service to handle more input formats correctly

* fix: route to local vision model provider when using ollama

* fix: improve temp file cleanup in image conversion

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>

Pistasha committed:
Fix: Correct spelling errors in texts and components (#2447)

* Update cosmosHelper.character.json

* Update cosmosHelper.character.json

* Update eternalai.character.json

* Update breadcrumb.tsx

* Update characters/cosmosHelper.character.json

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

azep-ninja committed:
add TG autonomous (#2338)

snobbee committed:
feat: add default logger level setting to set the logger level to display (#2451)

ai16z-demirix committed:
feat: adding tests for instagram client (#2454)

Co-authored-by: Sayo <hi@sayo.wtf>

Dimitrolito committed:
fix spelling error

Dimitrolito committed:
fix spelling error

Dimitrolito committed:
fix spelling error

savageops committed:
feat: create version.cmd for windows support (#2442)

* Create version.cmd

Initial Error Investigation
   - Encountered build error in client package
   - Error indicated `version.sh` script couldn't be executed on Windows.

Windows Script Compatibility Fix
   - Examined original `version.sh` shell script functionality
   - Created equivalent `version.cmd` batch script for Windows
   - Updated script to use Windows-compatible commands:

* Update client/version.cmd

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

---------

Co-authored-by: Sayo <hi@sayo.wtf>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

Dimitrolito committed:
fix spelling error

Dimitrolito committed:
fix spelling error

Dimitrolito committed:
fix spelling error

wtfsayo committed:
Merge branch 'develop' into fix/spelling

twilwa committed:
Revert "refactor: dockerize smoke tests (#2420)" (#2459)

This reverts commit 52565607a98c1b2f71fec380660bb3c83f0389c8.

thearyanag committed:
feat: Add features to the Solana Agent Kit  (#2458)

* wip: solana-agent-kit

* feat: swap

* feat: transfer

* feat: swap dao

* broken lock file

* lock file

* feat: add lend/stake/bounty/token info action from agent kit

* feat: agent kit pkg update

* add actions

* feat: Adds functions to the solana agent kit plugin

* feat: rename solana-agentkit to solana-agent-kit

wtfsayo committed:
Merge branch 'develop' into fix/spelling

wtfsayo committed:
chore: set openai as default provider (#2460)

wtfsayo committed:
Merge branch 'develop' into fix/spelling

dedyshkaPexto committed:
Typo fix in read.me

tcm390 committed:
fix: OPENAI provider being overwritten by LLAMA_LOCAL on pnpm start (#2465)

* revert

* fix onchainJson

* improve i6900 code

---------

Co-authored-by: Sayo <hi@sayo.wtf>

tcm390 committed:
Merge pull request #2464 from dedyshkaPexto/fix-typo

Typo fix in read.me

tcm390 committed:
Merge branch 'develop' into fix/spelling

tcm390 committed:
Merge pull request #2456 from Dimitrolito/fix/spelling

Fix spelling error

ai16z-demirix committed:
feat: adding tests for supabase adapter

carlos-cne committed:
fix React rules on AgentRoute component (#2413)

ai16z-demirix committed:
feat: adding advanced for supabase adapter

ai16z-demirix committed:
feat: adding config for sqlite- adapter tests

ai16z-demirix committed:
feat: adding tests for sqlite-adapter

ai16z-demirix committed:
test: removing sufficient vitest package from configuration

ai16z-demirix committed:
test: removing sufficient vitest package from configuration for adapter-supabase

ai16z-demirix committed:
Merge branch 'develop' into test/db-adapters

ai16z-demirix committed:
test: adding test configuration for adapter-redis

ai16z-demirix committed:
test: adding tests for adapter-redis

ai16z-demirix committed:
Merge branch 'develop' into tests/redis-adapter

suleigolden committed:
fix: resolve type error for children in animated.div component (#2462)

Co-authored-by: Sayo <hi@sayo.wtf>

B1boid committed:
fix swaps evm plugin (#2332)

wtfsayo committed:
Revert "Revert "refactor: dockerize smoke tests (#2420)" (#2459)"

This reverts commit e28353c63b7848e251cb3f6a5f54af590a21dcf4.

wtfsayo committed:
add openai var

twilwa committed:
Merge pull request #2472 from elizaOS/revert-2459-revert-2420-feat/dockerize-smoke-tests

chore: add openai env to smokeTests

shakkernerd committed:
Merge branch 'develop' into tests/redis-adapter

shakkernerd committed:
Merge pull request #2470 from ai16z-demirix/tests/redis-adapter

test: adding test configuration and tests for redis adapter

shakkernerd committed:
Merge branch 'develop' into test/db-adapters

shakkernerd committed:
Merge pull request #2468 from ai16z-demirix/test/db-adapters

feat: tests for supabase and sqlite db adapters

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: remove unused error var

shakkernerd committed:
fix: error logging and remove unused import

shakkernerd committed:
chore: console -> elizaLogger

shakkernerd committed:
fix: unused import and type errors

shakkernerd committed:
fix: lint command

shakkernerd committed:
chore: let -> const

shakkernerd committed:
Merge pull request #2474 from elizaOS/fixes

fix: linting errors

0xSero committed:
chore: remove eslint, prettier, tslint and replace with biome (#2439)

* remove eslint, and prettier. replace with biome-js

* full lint and biome switch

* lint check fix

* warn

* warn

* biome ci

* Update ci.yaml

---------

Co-authored-by: Sero <69639595+Seroxdesign@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

lispking committed:
feat: Sui supports the secp256k1/secp256r1 algorithms (#2476)

* feat: support secp256k1/secp256r1 for sui

* fix coderabbitai review

Olexandr88 committed:
Update README_DE.md (#2483)

sunsakis committed:
feat: Solana plugin improvement for flawless transfers (#2340)

* work in progress

* gods readme

* gods readme

* gods readme

* gods readme

* solana-plugin separate SPL and native transfers

* synced with twitter

* update README

* sbf readme

* update

* update readme

* added tech details

* added tech details

* simplified

* note

* original readme

* original readme

* original readme

* prepping for Eliza PR

* added similes and removed logging

* edited duplicate simile

* updated solana-plugin readme

* restored characters line

* added original README

eltociear committed:
chore: update createToken.ts (#2493)

creat -> create

Olexandr88 committed:
chore: corrected the link to the banner (#2491)

btspoony committed:
feat: introduce Dependency Injection to enhance developer experience (#2115)

* feat: Add @elizaos/plugin-di package with initial implementation

- Created a new plugin package `@elizaos/plugin-di` to extend the Eliza platform.
- Added essential files including `package.json`, `README.md`, and TypeScript configuration.
- Implemented core functionalities such as dependency injection using Inversify, action handling, and content evaluation.
- Introduced decorators for content properties and Zod schema integration.
- Included sample actions and evaluators to demonstrate plugin capabilities.
- Configured build and linting scripts for development workflow.

* feat: add tests for content decorators in plugin-di

- Introduced a new TypeScript configuration file `tsconfig.build.json` for build-specific settings.
- Updated `tsconfig.json` to enable experimental decorators and metadata emission.
- Added unit tests for content decorators, validating schema creation and property description loading.
- Implemented tests for the `property` decorator to ensure metadata storage functionality.

* feat: implement base injectable action class in plugin-di

- Removed export of providers from index.ts.
- Added a new file `baseInjectableAction.ts` that defines the `BaseInjactableAction` class, which serves as an abstract base for creating injectable actions.
- Introduced action options and methods for processing messages, validating content, and handling actions within the Eliza framework.
- Updated `index.ts` in actions directory to export the new base class.

* fix: correct export typo and refactor evaluators in plugin-di

- Fixed a typo in the export statement from `evaluators` to `evalutors` in `index.ts`.
- Added a new file `baseInjectableEvalutor.ts` to define the base class for injectable evaluators.
- Removed obsolete sample action and evaluator files to streamline the codebase.
- Updated `types.ts` to include the `ActionOptions` type definition for better action handling.

* feat: enhance plugin-di with new evaluators and action examples

- Fixed export typo in `index.ts` from `evalutors` to `evaluators`.
- Added `EvaluatorOptions` type definition in `types.ts` for better evaluator configuration.
- Introduced new sample action and evaluator files to demonstrate usage within the plugin.
- Implemented a base abstract class for injectable evaluators in `baseInjactableEvaluator.ts`.
- Updated `createPlugin` function to handle actions more effectively.
- Removed obsolete `index.ts` from the `evalutors` directory to clean up the codebase.

* feat: update CreateResource action to handle nullable content and refactor structure

- Modified the `execute` method in `BaseInjactableAction` and `CreateResourceAction` to accept nullable content.
- Introduced a new `CreateResourceContent` class to define the structure of the content for resource creation.
- Refactored the `CreateResourceAction` to utilize the new content class and improved validation logic.
- Enhanced the callback responses to handle cases where no content is provided, ensuring better error handling.

* feat: enhance plugin-di with new sample action, evaluator, provider, and plugin integration

- Introduced `CreateResourceAction` to manage resource creation with dependency injection.
- Added `SampleEvaluator` for evaluating important content in memory, implementing the `BaseInjactableEvaluator`.
- Created `SampleProvider` for data retrieval, adhering to the `InjectableProvider` interface.
- Registered new components with the global container for dependency injection.
- Defined a `samplePlugin` to encapsulate the new action, evaluator, and provider, facilitating resource management.

* refactor: update dependency injection for actions and providers

- Changed `CreateResourceAction` and `SampleEvaluator` to use `inRequestScope()` for better lifecycle management.
- Introduced dynamic data binding in `SampleProvider` with `toDynamicValue()` for asynchronous data retrieval.
- Updated `SampleProvider` to inject dynamic data and modified the `get` method to return a string representation of the shared and dynamic data.
- Adjusted import statements in `baseInjectableAction.ts` and `baseInjactableEvaluator.ts` to use `import type` for improved type safety.

* fix: improve null safety in normalizeCharacter function

- Updated the normalizeCharacter function to use optional chaining for plugin properties, enhancing null safety.
- Removed unnecessary return statements to streamline the code flow, ensuring that the function consistently returns the plugin object when valid.

* feat: apply di in agent index.ts

* feat: enhance error handling and null safety in plugin-di

- Updated the `createPlugin` function to improve error handling for providers, actions, and evaluators by logging errors and filtering out undefined values.
- Implemented optional chaining in the `buildContentOutputTemplate` function to enhance null safety when accessing examples.
- Added a new test suite for normalizing characters, ensuring the integrity of plugins and their properties.

* docs: update README.md for Dependency Injection Plugin

* fix: correct typo in comment for character normalization in index.ts

* feat: add external dependencies for plugin-di

- Included "inversify", "reflect-metadata", "zod", and "uuid" as external modules in the tsup configuration to enhance functionality and support for dependency injection and validation.

* fix: pnpm-lock.yaml

* refactor: remove ExtendedPlugin type and update createPlugin return type

- Removed the ExtendedPlugin type definition from types.ts to simplify the type structure.
- Updated the createPlugin function in plugin.ts to return a Plugin type instead of ExtendedPlugin, reflecting the changes in the type definitions.

* Update packages/plugin-di/src/actions/baseInjectableAction.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* Update packages/plugin-di/src/actions/baseInjectableAction.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* Update packages/plugin-di/src/actions/baseInjectableAction.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* Update package.json

* Refactor: fix typo, rename Injactable to Injectable in plugin-di

This commit updates the naming convention from `Injactable` to `Injectable` across the plugin-di package, enhancing clarity and consistency. Changes include:
- Renaming classes and interfaces in README.md, types.ts, and example files.
- Introducing a new `BaseInjectableEvaluator` class to replace the previous `BaseInjactableEvaluator`.
- Updating imports and references throughout the codebase to reflect the new naming.

These modifications improve the overall readability and maintainability of the code.

* chore: update dependencies and fix typos in plugin-di

- Removed outdated dependencies from pnpm-lock.yaml.
- Updated zod version from ^3.24.1 to 3.23.8 in package.json and pnpm-lock.yaml.
- Fixed import typos in samplePlugin.ts and README.md, changing 'sampleEvalutor' to 'sampleEvaluator'.
- Added new sampleEvaluator class in sampleEvaluator.ts for improved functionality.
- Enhanced test coverage in normalizeCharacter.test.ts to handle edge cases.

These changes improve dependency management and code clarity.

* fix: improve plugin normalization and export syntax in plugin-di

- Updated the export syntax in index.ts for better clarity by removing quotes around "symbols".
- Enhanced the normalization process in charactor.ts to log warnings when some plugins fail to normalize, improving error handling and debugging.

These changes enhance code readability and robustness in plugin handling.

* Apply suggestions from code review

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* refactor: enhance error handling and code clarity in plugin-di

- Improved error handling in CreateResourceAction by logging a specific error message when no content is provided.
- Refactored plugin factory functions to utilize a new helper function, getInstanceFromContainer, for better readability and maintainability.
- Streamlined the normalization process for providers, actions, and evaluators, ensuring consistent error logging and filtering of undefined values.

These changes enhance the robustness and clarity of the plugin handling process.

* Apply suggestions from code review

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* feat: resolve coderabbitai

* Update packages/plugin-di/src/_examples/sampleAction.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* chore: update pnpm-lock.yaml

* fix: pnpm lock

* fix: update pnpm-lock and revert wrong modifications in default samplePlugin

* chore: change version to 0.1.8+build.1 like others

* fix: pnpm-lock

* fix: pnpm-lock.yaml

* fix: pnpm-lock.yaml

* chore: Update version and README for @elizaos/plugin-di

* refactor: update access modifiers in BaseInjectableAction and clean up sampleEvaluator imports

- Changed private properties to protected in BaseInjectableAction for better inheritance support.
- Cleaned up imports in sampleEvaluator by removing unused State import and standardizing string quotes.

* refactor: change return type of processMessages callback in BaseInjectableAction to Promise<any | null>

- Updated the return type of the callback in the BaseInjectableAction class to allow for more flexible handling of results, returning null instead of void.

* chore: update pkg

---------

Co-authored-by: Sayo <hi@sayo.wtf>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

Olexandr88 committed:
Update README_JA.md (#2490)

Olexandr88 committed:
Update README_CN.md (#2489)

Co-authored-by: Sayo <hi@sayo.wtf>

Daulox92 committed:
Update README.md (#2495)

KoZivod88074 committed:
Improve error handling and typing in chat-bubble component (#2485)

* Update chat-bubble.tsx

* Update version.sh

ai16z-demirix committed:
feat: adding test configuration and test coverage for binance plugin  (#2482)

* test: adding test configuration for plugin-binance

* test: adding account tests for plugin-binance

* test: adding price tests for plugin-binance

* test: adding trade tests for plugin-binance

---------

Co-authored-by: Sayo <hi@sayo.wtf>

tercel committed:
feat: Implement character loading from multiple paths and enhance API… (#2365)

* feat: Implement character loading from multiple paths and enhance API for character management

* fix: Update character handling to prioritize username and email fields

- Modified the AgentRuntime class to use character.username instead of character.name for user identification.
- Enhanced user account creation logic to utilize character.email and character.name for better user data management.
- Updated Character type definition to include optional email field.

leeran7 committed:
feat: Load multiple characters from a single remote url (#2475)

* Feat: Load multiple characters from a single remote url

* Chore: Added fetch try/catch

* Chore: added parenthesis around s in error message

---------

Co-authored-by: Sayo <hi@sayo.wtf>

azep-ninja committed:
feat: Discord autonomous agent enhancement (#2335)

* add Discord autonomous agent

* sync fixes

* fix redundancy

* fix redundancy

* fix

tcm390 committed:
support multimedia

tcm390 committed:
destroy file system after sending media

tcm390 committed:
Replace type assertions

tcm390 committed:
Replace type assertions

tcm390 committed:
Merge pull request #2510 from elizaOS/tcm-telegram-media

feat: Add Extra Multimedia Support for Telegram Client

ai16z-demirix committed:
test: api timeout handling for plugin-binance (#2504)

* test: api timeout handling for plugin-binance

* test: api timeout handling for plugin-binance - PR feedback

---------

Co-authored-by: Sayo <hi@sayo.wtf>

AIFlowML committed:
feat: add support for NVIDIA inference for ElizaOS (#2512)

* Adding support for NVIDIA inference for ElizaOS

* removed wrong image generation

* Fixed wrong indentation

---------

Co-authored-by: Sayo <hi@sayo.wtf>

0xrubusdata committed:
feat(plugin-openai): add OpenAI integration for text generation (#2463)

* feat: add plugin-openai for text generation using OpenAI API (#1234)

* fix: apply code review suggestions from coderabbitai

* add remaining updates for plugin

* Refactor: Centralize OpenAI action utilities in action.ts

- Introduced shared constants (DEFAULT_MODEL, DEFAULT_MAX_TOKENS, etc.).
- Added reusable functions:
  - validatePrompt: Validates text input for OpenAI actions.
  - validateApiKey: Ensures API key is present.
  - callOpenAiApi: Handles HTTP requests with timeout and error handling.
  - buildRequestData: Creates structured request payloads.
- Updated all action files to leverage shared logic.
- Improved maintainability and consistency across all OpenAI-related actions.

* feat(plugin-openai): finalize plugin setup and configuration

- Updated  to enhance terminal output with plugin details and accounts (Twitter, GitHub).
- Adjusted  scripts and dependencies for compatibility and functionality.
- Added a localized  for specific TypeScript configurations within the plugin.
- Improved initialization and logging to streamline the startup process.

This commit completes the setup for the OpenAI plugin and ensures integration with the framework.

---------

Co-authored-by: 0xrubusdata <0xrubusdatas@gmail.com>
Co-authored-by: Sayo <hi@sayo.wtf>
Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>

sweetmantech committed:
AgentKit - Default Agent Config (#2505)

* I add files for plugin-agentkit

* instructions added to gitignore.

* I update package name.

* I add updated files.

* I add ENV requirements to prevent default Eliza bugs.

* rollback index

* simple agentkit import.

* I load existing wallet data for provider.

* remove dev logging

* Update pnpm-lock.yaml

* update .env.example and support non 'base-sepolia'

* Update pnpm-lock.yaml

* Update package lock for smoke tests.

* I rollback changes to pnpm-lock file.

* fixing smoke tests for pnpm missing

* rollback changes to agent start file.

* I rollback changes to smoke test.

* Update pnpm-lock.yaml

* packages/plugin-agentkit/README.md - package name
    - actual: @ai16z/plugin-agentkit
    - required: @elizaos/plugin-agentkit

* add example network for agentkit.

* update plugin-agentkit README package name to correct ORG prefix.

* agent/src/index.ts - default plugin
    - actual:agentKitPlugin missing from AgentRuntime.plugins
    - required: agentKitPlugin added to AgentRuntime.plugins

* stylish build logging.

* wallet read from ENV instead of function variable.

* remove unintended changes to agent index.

---------

Co-authored-by: Sayo <82053242+wtfsayo@users.noreply.github.com>
Co-authored-by: Sayo <hi@sayo.wtf>

eltociear committed:
docs: add docs/README_JA.md (#2515)

Co-authored-by: Sayo <hi@sayo.wtf>

visionpixel committed:
fix: use header key from api config (#2518)

Co-authored-by: Marcel Rableski <marcel.rableski@sva.de>
Co-authored-by: Sayo <hi@sayo.wtf>

ai16z-demirix committed:
test: plugin-tee - adjusting project structure and new tests (#2508)

* test: plugin-tee - adjusting project strucutre and new tests

* test: plugin-tee - ficed test structure

---------

Co-authored-by: Sayo <hi@sayo.wtf>

emeieiron committed:
Merge branch 'develop' into develop

odilitime commented:
Hello,

We are changing our plugin development strategy to be more scalable. We have moved the plugins out into their own repos and we're looking for people to either maintain those or own them on their own Github.

Please make your code changes to this repo: https://github.com/elizaos-plugins/plugin-sui

If you'd like to be a maintainer,  file an issue in the plugin repo and join our Discord https://discord.gg/elizaos to coordinate.

If you'd like to control the plugin on your own Github, please add an issue to the plugin repo pointing to your repo, and add a modification to the registry. Submit a PR to edit the registry here: https://github.com/elizaos-plugins/registry

Closing this PR for now. Let us know if you have any questions.
